src/modules/fw_path_navigation/FixedwingPositionControl.cpp

```diff
@@ -696,39 +696,42 @@ FixedwingPositionControl::updateManualTakeoffStatus()
 	}
 }
 
-void
-FixedwingPositionControl::set_control_mode_current(const hrt_abstime &now)
+FW_POSCTRL_MODE
+FixedwingPositionControl::set_control_mode_current(const hrt_abstime &now, const FW_POSCTRL_MODE &current_mode)
```

Comment:
How about 

```suggestion
FixedwingPositionControl::set_control_mode_current(const hrt_abstime &now, const FW_POSCTRL_MODE &previous_mode)
```

Reply:
To make it more explicit what's the old (input) and what's the new (output) mode.